packages/nvidia-container-toolkit/nvidia-container-toolkit-config-k8s

```diff
@@ -1,14 +1,39 @@
 [required-extensions]
 nvidia-container-runtime = "v1"
-std = { version = "v1", helpers = ["default"] }
+kubelet-device-plugins = "v1"
+std = { version = "v1", helpers = ["default", "is_array"] }
 
 +++
 ### generated from the template file ###
 accept-nvidia-visible-devices-as-volume-mounts = {{default true settings.nvidia-container-runtime.visible-devices-as-volume-mounts}}
 accept-nvidia-visible-devices-envvar-when-unprivileged = {{default false settings.nvidia-container-runtime.visible-devices-envvar-when-unprivileged}}
-
 [nvidia-container-cli]
 root = "/"
 path = "/usr/bin/nvidia-container-cli"
 environment = []
 ldconfig = "@/sbin/ldconfig"
+
+[nvidia-container-runtime]
+{{#if settings.kubelet-device-plugins.nvidia.device-list-strategy}}
+{{~#if (is_array settings.kubelet-device-plugins.nvidia.device-list-strategy) ~}}
+{{~#if (eq settings.kubelet-device-plugins.nvidia.device-list-strategy.[0] "cdi-cri") ~}}
+mode="cdi"
```

Comment:
Is it expected that "cdi-cri" will always be the first item of the array? What is the behavior if we have multiple items in the array, and what if "cdi-cri" is the second item in the array? 

May not be blocking. But it may be good to implement some helper like "has/contains" - similar to what they did in the device plugin - https://github.com/NVIDIA/k8s-device-plugin/blob/6f41f70c43f8da1357f51f64cf60431acc74141f/deployments/helm/nvidia-device-plugin/templates/_helpers.tpl#L178.

Also a note here - I was going to warn the index out of bound concern, but looks like the `if` helper does check for empty list and treat it as false, so using "0" is safe here.

Reply:
Regarding the conversation about custom helper, we had one, we were advice not to add it at least for this case:

https://github.com/bottlerocket-os/bottlerocket-core-kit/pull/502